pkgs/by-name/cr/cryptpad/package.nix

```diff
@@ -18,18 +18,55 @@ let
   # nix version of install-onlyoffice.sh
   # a later version could rebuild from sdkjs/web-apps as per
   # https://github.com/cryptpad/onlyoffice-builds/blob/main/build.sh
-  onlyoffice_build =
+  onlyoffice_oldfetch =
     rev: hash:
     fetchFromGitHub {
       inherit rev hash;
       owner = "cryptpad";
       repo = "onlyoffice-builds";
     };
+
+  # New method for v7+ versions that use ZIP releases
+  onlyoffice_newfetch =
+    version: hash:
+    fetchurl {
+      url = "https://github.com/cryptpad/onlyoffice-editor/releases/download/${version}/onlyoffice-editor.zip";
+      inherit hash;
+    };
+
   onlyoffice_install = oo: ''
     oo_dir="$out_cryptpad/www/common/onlyoffice/dist/${oo.subdir}"
-    cp -a "${onlyoffice_build oo.rev oo.hash}/." "$oo_dir"
-    chmod -R +w "$oo_dir"
-    echo "${oo.rev}" > "$oo_dir/.commit"
+    ${
+      if oo ? "version" then
+        ''
+          # New method for v7+ versions
+          mkdir -p "$oo_dir"
+          cd "$oo_dir"
+          unzip ${onlyoffice_newfetch oo.version oo.hash}
+          echo "${oo.version}" > "$oo_dir/.version"
+
+          # Clean up help files and dictionaries as per upstream (using relative paths since we're in $oo_dir)
+          rm -rf web-apps/apps/documenteditor/main/resources/help
+          rm -rf web-apps/apps/presentationeditor/main/resources/help
+          rm -rf web-apps/apps/spreadsheeteditor/main/resources/help
+          ${lib.optionalString (oo.subdir == "v7") ''
```

Comment:
we don't want to remove any doc for v8 (install-onlyoffice.sh only does this for old versions)